### PR TITLE
Master

### PR DESCRIPTION
**Pull request by zhuscat**
有几句话没有翻译。

1. Together with intersection types, we can make some pretty mind-bending types

不清楚 pretty mind-bending types 是不是我那么翻译的。

2. but it’s much more obvious when you forget it

3. 那么s将具有一个赶写的类型改为将具有一个真实的类型并且你会得到一个错误

4. 字符器字面量类型改变字符串字面量类型

### CODE REVIEW

**Comment by zhongsp** [comment]
@zhuscat Thanks! 是否可以修改commit，不提交另3个文件的改动呢？

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
-node_modules/
-.DS_Store
-_book
+node_modules/
+.DS_Store
+_book
```
Comment (zhongsp) [inline .gitignore]: How about leaving this file unchanged?

File: doc/release-notes/TypeScript 2.0.md [hunk 2]
```diff
@@ -1,877 +1,877 @@
...
-    code?: string;
-    path?: string;
-    stack?: string;  // OK
-}
-```
-
-## 新编译参数`--declarationDir`
-
-`--declarationDir`可以使生成的声明文件和JavaScript文件不在同一个位置中。
+## Null和undefined类型
```
Comment (zhongsp) [inline doc/release-notes/TypeScript 2.0.md]: How about leaving this file unchanged?

File: gulpfile.js [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
-
-  return tsResult.js.pipe(gulp.dest('dist/typescript'));
-});
-
-gulp.task('watch', function() {
-  gulp.watch(['ts/**/*.ts'], ['typescript']);
-});
-
-gulp.task('default', ['typescript', 'watch']);
+var gulp = require('gulp'),
```
Comment (zhongsp) [inline gulpfile.js]: How about leaving this file unchanged?

Comment (zhuscat) [inline gulpfile.js]: @zhongsp 我把 clone 下来的文件替换进去也没用，虽然文件内容是一样的，但是却说改变了。

Comment (zhuscat) [inline gulpfile.js]: 要怎么消除呢

**Comment by zhongsp** [comment]
Edited your commit and merged on command line. Thanks @zhuscat .

**Pull request closed without merging**